pkg/skaffold/event/v2/status_check.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/constants"
+	proto "github.com/GoogleContainerTools/skaffold/proto/v2"
+)
+
+func ResourceStatusCheckEventCompleted(r string, ae proto.ActionableErr) {
+	if ae.ErrCode != proto.StatusCode_STATUSCHECK_SUCCESS {
+		resourceStatusCheckEventFailed(r, ae)
+		return
+	}
+	resourceStatusCheckEventSucceeded(r)
+}
+
+func resourceStatusCheckEventSucceeded(r string) {
+	handler.handleStatusCheckSubtaskEvent(&proto.StatusCheckSubtaskEvent{
+		TaskId:     fmt.Sprintf("%s-%d", constants.StatusCheck, handler.iteration),
+		Resource:   r,
+		Status:     Succeeded,
+		Message:    Succeeded,
+		StatusCode: proto.StatusCode_STATUSCHECK_SUCCESS,
+	})
+}
+
+func resourceStatusCheckEventFailed(r string, ae proto.ActionableErr) {
+	handler.handleStatusCheckSubtaskEvent(&proto.StatusCheckSubtaskEvent{
+		TaskId:        fmt.Sprintf("%s-%d", constants.StatusCheck, handler.iteration),
+		Resource:      r,
+		Status:        Failed,
+		StatusCode:    ae.ErrCode,
```

Comment:
we don't add a `Message` for failures? that's embedded in the actionable error then?

Reply:
mhm, I'm not 100% on the original intent as this is from v1, but I think it's basically that in the case of a non-failure, we still want a message, and in the case of  a failure, we put the message in the actionable error